wgpu-hal/src/metal/mod.rs

```diff
@@ -388,8 +390,23 @@ impl crate::Queue<Api> for Queue {
     }
 
     unsafe fn get_timestamp_period(&self) -> f32 {
-        // TODO: This is hard, see https://github.com/gpuweb/gpuweb/issues/1325
-        1.0
+        if !self.shared.private_caps.supports_timestamp_period {
+            return 1.0;
+        }
+
+        use objc::{msg_send, runtime::Object, sel, sel_impl};
+        let timestamps0 = [std::ptr::null_mut::<Object>(); 2];
+        let device = self.shared.device.lock().as_ptr();
+        let () = msg_send![device, sampleTimestamps:&timestamps0[0] gpuTimestamp:&timestamps0[1]];
```

Comment:
should use mutable borrows

Reply:
timestamps0 is a `[*mut Object; 2]` array, so, `&timestamps0[x]` already a mutable pointer for objc